### DISCUSSION TITLE
How to extract the pre-final layer output separately in yolo v8n/x ? ( or extract features from specific layers in YOLOv8 )

### DISCUSSION THREAD

**Discussion by Akshaykushawaha** [ultralytics/ultralytics#7888] [Q&A]
## Need to extract the features from the pre-final layer in yolo v8 in general object detection model.
I asked this question before and @glenn-jocher suggested that i would be better off getting a detailed reply here.
According to him i need to:
> To access the forward method in YOLOv8, you'll need to look into the model's architecture file, typically named something like model.py or similar. Within this file, locate the forward function. To extract the pre-final layer, you'll need to identify the index of this layer and then insert a return statement for that layer's output.

i am unable to find a model.py or a suitable forward function to edit, i saw files like:
Ultralytics repo --> ultralytics --> models --> yolo --> detect --> train.py
and
Ultralytics repo --> ultralytics --> nn --> tasks.py

But i cannot find any other suitable files. Basically i haven't progressed at all, Kindly help me with the place/file to modify and i'll do accordingly.

Thanks in Advance!

**Comment by glenn-jocher** [comment]
Just use model.embed() for creating embeddings

**Reply by Akshaykushawaha** [reply]
okay, will try this and get back.

**Reply by pderrenger** [reply]
@Akshaykushawaha sounds good! If you encounter any issues or have further questions, feel free to reach out. Good luck with your feature extraction! 👍
